src/SWP/Component/MultiTenancy/Factory/TenantFactory.php

```diff
@@ -13,31 +13,73 @@
  */
 namespace SWP\Component\MultiTenancy\Factory;
 
+use SWP\Component\Common\Generator\GeneratorInterface;
+use SWP\Component\MultiTenancy\Model\TenantInterface;
+use SWP\Component\MultiTenancy\Repository\OrganizationRepositoryInterface;
+use SWP\Component\Storage\Factory\FactoryInterface;
+
 /**
  * Class TenantFactory.
  */
 class TenantFactory implements TenantFactoryInterface
 {
     /**
-     * @var string
+     * @var FactoryInterface
      */
-    protected $className;
+    protected $decoratedFactory;
+
+    /**
+     * @var GeneratorInterface
+     */
+    protected $generator;
+
+    /**
+     * @var OrganizationRepositoryInterface
+     */
+    protected $organizationRepository;
 
     /**
      * TenantFactory constructor.
      *
-     * @param string $className
+     * @param FactoryInterface                $decoratedFactory
+     * @param GeneratorInterface              $generator
+     * @param OrganizationRepositoryInterface $organizationRepository
      */
-    public function __construct($className)
-    {
-        $this->className = $className;
+    public function __construct(
+        FactoryInterface $decoratedFactory,
+        GeneratorInterface $generator,
+        OrganizationRepositoryInterface $organizationRepository
+    ) {
+        $this->decoratedFactory = $decoratedFactory;
+        $this->generator = $generator;
+        $this->organizationRepository = $organizationRepository;
     }
 
     /**
      * {@inheritdoc}
      */
     public function create()
     {
-        return new $this->className();
+        /** @var TenantInterface $tenant */
```

Comment:
This is component so - yes, it should be possible as it can be used not only in SWP